soc_026_rw0_global_gender_gap/soc_026_rw0_global_gender_gap.py

```diff
@@ -0,0 +1,200 @@
+    # rename columns
+    df_concat.columns = processed_cols
+    # add clean data to overall dataframe for main index
+    data_main_index = pd.concat([data_main_index, df_concat]).reset_index(drop=True)
+
+
+def process_subindex(subindex, l_r, page):
+    if l_r=='left':
+        start_i = 0
+    else:
+        start_i = 6
+    cols_sub = [f'{subindex}_subindex_rank', 'country', f'{subindex}_subindex_score']
+    df_subindex = pd.DataFrame(columns=cols_sub)
+    for year, df_list in df_subindex_dict.items():
+        df_a = df_list[page].iloc[:, start_i:start_i+3] #first half
+        df_a.columns = cols_sub
+        df_b = df_list[page].iloc[:, start_i+3:start_i+6] #second half
+        df_b.columns = cols_sub
+        df_concat = pd.concat([df_a, df_b]).reset_index(drop=True) #concat
+        df_concat['year'] = year #create a column for year
+        # replace , with . to properly format decimals in Score column
+        df_concat[[cols_sub[2]]] = df_concat[[cols_sub[2]]].replace(',','.', regex=True)
+        df_concat = df_concat.dropna(subset = ['country']) #drop NA
+        # the source data has mistakenly replaced , with . in some country names
+        # if there is a . after the first word of a country name, replace it with a comma
+        df_concat['country'] = df_concat['country'].apply(lambda x: x.replace(str(x).split(' ')[0], str(x).split(' ')[0].replace('.', ',')))
+        # replace inconsistent country names
+        for country in name_mapping.keys():
+            df_concat['country'] = np.where(df_concat['country']==country, name_mapping[country], df_concat['country'])
+        # drop asterisks from new countries in country column
+        df_concat[['country']] = df_concat[['country']].replace('\*','', regex=True)
+        df_subindex = pd.concat([df_subindex, df_concat]).reset_index(drop=True)
+    return df_subindex
+
+
+econ_df = process_subindex('economic_participation_and_opportunity', l_r='left', page=0)
+edu_df = process_subindex('educational_attainment', l_r='right', page=0)
+health_df = process_subindex('health_and_survival', l_r='left', page=1)
+political_df = process_subindex('political_empowerment', l_r='right', page=1)
+
+# define the list of dataframes to merge
+merge_list = [econ_df, edu_df, health_df, political_df]
+df_processed = data_main_index.copy()
+for df in merge_list:
+    df_processed = pd.merge(df_processed, df, on = ['year', 'country'], how = 'outer')
+        
+# read in existing dataframe so we can merge the new data with the old data
+api_key = os.getenv('CARTO_WRI_RW_KEY')
+username = os.getenv('CARTO_WRI_RW_USER')
+q = 'SELECT * FROM soc_026_gender_gap_index_1'
+url = 'https://wri-rw.carto.com/api/v2/sql'
+r = requests.get(url, params={'api_key': api_key, 'q': q}).text
+df_dict = json.loads(r)
+df_carto = pd.DataFrame(df_dict['rows'])
+
+# Merge new years with old
+frames_carto_upload = [df_carto, df_processed]
+df_carto_upload = pd.concat(frames_carto_upload).reset_index(drop=True)
```

Comment:
```suggestion
df_carto_upload = pd.concat(frames_carto_upload).reset_index(drop=True)

#drop cartodb_id so that it can be reassigned during upload
df_carto_upload = df_carto_upload.drop('cartodb_id', axis=1)
```